addons/mrp/static/src/js/cost_structure_report.js

```diff
@@ -0,0 +1,64 @@
+odoo.define('report.bom_cost_structure', function (require) {
+'use strict';
+var core = require('web.core');
+var ReportAction = core.action_registry.get('report.client_action');
+
+
+var BomReportAction = ReportAction.extend({
+    init: function (parent, action, options){
+        var option = _.extend({}, options, {
+                report_url: '/report/html/' + action.report_name + '/' + action.context.active_id,
+                report_name: action.report_name,
+                report_file: action.report_file,
+                data: action.data,
+                context: action.context,
+                name: action.name,
+                display_name: action.display_name,
+            });
+        this._super(parent, action, option);
+    },
```

Comment:
why?

Reply:
because of report.client_action hardcoded [here](https://github.com/odoo/odoo/blob/master/addons/web/static/src/js/report/qwebactionmanager.js#L87). so server [return](https://github.com/odoo/odoo/pull/21091/files/0ab2f6b5e83230468a1b59e00a25d2d151115bac#diff-a654ed968da44a062292dbac946c69a6R172) client action and use it as report client action